### PR TITLE
Enforce right-alignment for cell in markdown tables

### PR DESCRIPTION
**Pull request by gnestor**
Fixes https://github.com/jupyterlab/jupyterlab/issues/2151

### CODE REVIEW

**Comment by takluyver** [comment]
I'm wary of adding `!important` rules because they're hard for anyone else to override. Are we so confident in our opinion of how tables should look that we're willing to more-or-less force it on everyone?

cc @ellisonbg

**Comment by gnestor** [comment]
@takluyver I suppose not because in https://github.com/jupyterlab/jupyterlab/issues/2151, @tdaff was actually looking to override the jupyterlab default styles for tables. We do want consistency across notebook and lab though and this would enforce that in notebook. I just don't understand why they're rendering differently given that they're both using marked... I suppose that it could be related to version inconsistency:

* lab: `"marked": "^0.3.6"`
* notebook: `"marked": "~0.3"`

**Comment by takluyver** [comment]
I've moved this to 'backlog' so we can get on with 5.1, but left it open in case people want to work out where the inconsistency is coming from and what we can do about it.

**Comment by gnestor** [comment]
Ya, I've explored the marked repo and there haven't been any changes to the markdown table style for 4 years, so the cause must be a discrepancy in the way marked is used in notebook vs. lab.

**Comment by datasouvik** [comment]
Hi gentlemen,

Has this issue been resolved? Do we now get the table content as left-aligned or centrally aligned by default, instead of the right alignment?

Thanks,
Souvik

**Comment by gnestor** [comment]
I have just updated this PR to prevent marked from adding inline style for table cells. As a result, there will be no need for `!important` and this will allow users to specify custom styles in custom.css.

**Comment by gnestor** [comment]
@takluyver Care to review/merge?

File: notebook/static/notebook/js/outputarea.js [hunk 1]
```diff
@@ -701,7 +701,16 @@ define([
...
+        // Prevent marked from returning inline styles for table cells
+        var renderer = new marked.Renderer();
+        renderer.tablecell = function (content, flags) {
+          var type = flags.header ? 'th' : 'td';
+          var start_tag = '<' + type + '>';
+          var end_tag = '</' + type + '>\n';
+          return start_tag + content + end_tag;
+        };
+        marked(text, { renderer: renderer }, function (err, html) {
+        // marked(text, function (err, html) {
```
Comment (takluyver) [inline notebook/static/notebook/js/outputarea.js]: Do you want to clear away these commented out lines now that the new code is working?

**Comment by gnestor** [comment]
@takluyver Thanks, merge away!

**Pull request merged**